scripts/deploy.sh

```diff
@@ -48,8 +49,13 @@ if [[ "${DEPLOY_AIO}" == "yes" ]]; then
   if [ -n "${DATA_DISK_DEVICE}" ]; then
     export BOOTSTRAP_OPTS="${BOOTSTRAP_OPTS} bootstrap_host_data_disk_device=${DATA_DISK_DEVICE}"
   fi
+
   # force the deployment of haproxy for an AIO
   export DEPLOY_HAPROXY="yes"
+
+  # Enable host security hardening for AIO builds
+  export DEPLOY_HARDENING="yes"
```

Comment:
Well this feature makes a whole load of changes that, as I understand it, will only be used by a limited number of customers. So we'll essentially be testing an edge case deployment as in all our standard gate job with this setting. In additional anyone wanting to do any AIO testing without this hardening stuff now has to manually edit the script to make that happen.

The only things that should be hardcoded in the AIO section are those things that are required to make the AIO work, e.g. haproxy.